GoogleForCreators/web-stories-wp
# Refactored workspace footer to separate carousel from rest

## Context

This is a partial for #9641. It's only a refactoring of existing architecture - this PR does not change any existing functionality or add new functionality.

The sole purpose of this is to introduce a new abstract element in the hierarchy - the _workspace footer_. Before we called the whole footer below the canvas the _"carousel"_, even though it consisted of other menus and buttons and the actual carousel only was a single part of it.

Now, only the middle part of the new workspace footer is considered the carousel, and the menus and buttons are simply members of the workspace footer alongside the carousel.

## User-facing changes

None

## Testing Instructions

This PR can be tested by following these steps:

1. Verify that carousel, checklist, keyboard shortcut menu, grid view, and zoom still look and function exactly as they do now.

## Checklist




---

Partially fixes #9641
